backend/api/Areas/Admin/Controllers/AgencyController.cs

```diff
@@ -118,11 +131,18 @@ public IActionResult AddAgency([FromBody] Model.AgencyModel model)
         [ProducesResponseType(typeof(Model.AgencyModel), 200)]
         [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
         [SwaggerOperation(Tags = new[] { "admin-agency" })]
-        public IActionResult UpdateAgency([FromBody] Model.AgencyModel model)
+        public async Task<IActionResult> UpdateAgencyAsync([FromBody] Model.AgencyModel model)
         {
             var entity = _mapper.Map<Entity.Agency>(model);
             _pimsAdminService.Agency.Update(entity);
```

Comment:
I'm pretty against using side effects to modify values like this (in this case the list of users) can we return something instead?

Reply:
Technically the DAL is doing more than it should.  All it cares about is the PIMS DB.  I've added returning the affected users so that we can keep Keycloak in sync.

In an ideal solution I would through dependency injection (DI) add an interface in the DAL that subsequently informs DI interface that an update has been made to this agency and allow DI to figure out how to talk to other services like Keycloak.  But the time to do this is extreme.  

So I've gone the functional route.  I've let the Controller handle the updating of Keycloak after it receives a list of affected users.

Reply:
All I'm asking is that you return an object and operate on that rather then depending on your function to modify values in one of the function parameters.

If for some reason you need to do this, then add an out variable so that at least it is clear which values the function modifies.